### PR TITLE
[GLUTEN-1246][CORE][Fix] Fix allowPrecisionLoss false mode and enable some test suites

### PR DESCRIPTION
**Pull request by jinchengchenghh**

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala [hunk 1]
```diff
@@ -299,13 +303,20 @@ object ExpressionConverter extends Logging {
             attributeSeq),
           u)
       case b: BinaryExpression =>
+        b match {
+          case arithmetic: BinaryArithmetic if isDecimalArithmetic(arithmetic)
+            && !conf.decimalOperationsAllowPrecisionLoss =>
```
Comment (jinchengchenghh) [inline gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:308]: @zzcclp Does CH backend support decimalOperationsAllowPrecisionLoss false mode? Default is true

Comment (zzcclp) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:308]: @taiyang-li please help to check this, thanks.

Comment (taiyang-li) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:308]: @jinchengchenghh @zzcclp  CH don't support this config, and its default behavior is disabling precision loss.

Comment (taiyang-li) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:308]: I also noticed that decimal type promotion is different 
 in CH and spark/hive: 
- spark/hive: https://cwiki.apache.org/confluence/download/attachments/27362075/Hive_Decimal_Precision_Scale_Support.pdf
- CH:  https://clickhouse.com/docs/en/sql-reference/data-types/decimal#operations-and-result-type

Comment (jinchengchenghh) [reply gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:308]: So I can put this code in gluten-core.
I implement same decimal type promotion in velox arithmetic.

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala [hunk 2]
```diff
@@ -49,17 +53,65 @@ object ExpressionConverter extends Logging {
     }
   }
 
+  private def getNewPrecisionScale(dec: Decimal): (Integer, Integer) = {
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:56]: Can we add some description for this function?

File: gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala [hunk 3]
```diff
@@ -49,17 +53,65 @@ object ExpressionConverter extends Logging {
...
+        if (arr(i) != 0) {
+          newScale = arr.length - dotIndex - 1 - i
+          break
+        }
+      }
+    }
+    (dotIndex + newScale, newScale)
+  }
+
+  private def rescaleLiteral(arithmeticExpr: BinaryArithmetic): Unit = {
```
Comment (rui-mo) [inline gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala:74]: Same for this one.

File: gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala [hunk 4]
```diff
@@ -43,17 +43,14 @@ class VeloxTestSettings extends BackendTestSettings {
...
-      "SPARK-36926: decimal average mistakenly overflow"
+      "SPARK-32038: NormalizeFloatingNumbers should work on distinct aggregate"
     )
 
   enableSuite[GlutenCastSuite]
     .exclude(
       "Process Infinity, -Infinity, NaN in case insensitive manner" // +inf not supported in folly.
     )
     // decimal failed ut
-    .exclude("Fast fail for cast string type to decimal type")
```
Comment (rui-mo) [inline gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:56]: Can we rebase to main and enable all the left varchar-to-decimal unit tests?

Comment (jinchengchenghh) [reply gluten-ut/spark32/src/test/scala/io/glutenproject/utils/velox/VeloxTestSettings.scala:56]: Fixed all the comments

**Pull request merged**